web/src/routes/(user)/trash/+page.svelte

```diff
@@ -0,0 +1,112 @@
+  import LinkButton from '$lib/components/elements/buttons/link-button.svelte';
+  import { AssetStore } from '$lib/stores/assets.store';
+  import { api, TimeBucketSize } from '@api';
+  import DeleteOutline from 'svelte-material-icons/DeleteOutline.svelte';
+  import HistoryOutline from 'svelte-material-icons/History.svelte';
+  import type { PageData } from './$types';
+  import { featureFlags } from '$lib/stores/server-config.store';
+  import { goto } from '$app/navigation';
+  import empty3Url from '$lib/assets/empty-3.svg';
+  import ConfirmDialogue from '$lib/components/shared-components/confirm-dialogue.svelte';
+
+  export let data: PageData;
+
+  $: $featureFlags.trash || goto(AppRoute.PHOTOS);
+
+  const assetStore = new AssetStore({ size: TimeBucketSize.Month, isTrashed: true });
+  const assetInteractionStore = createAssetInteractionStore();
+  const { isMultiSelectState, selectedAssets } = assetInteractionStore;
+  let isShowEmptyConfirmation = false;
+
+  const handleEmptyTrash = async () => {
+    isShowEmptyConfirmation = false;
+    try {
+      await api.assetApi.updateTrash({ updateTrashDto: { deleteAll: true } });
+
+      notificationController.show({
+        message: `Empty trash initiated. Refresh the page to see the changes`,
+        type: NotificationType.Info,
+      });
+    } catch (e) {
+      handleError(e, 'Error emptying trash');
+    }
+  };
+
+  const handleRestoreTrash = async () => {
+    try {
+      await api.assetApi.updateTrash({ updateTrashDto: { restoreAll: true } });
+
+      notificationController.show({
+        message: `Restore trash initiated. Refresh the page to see the changes`,
+        type: NotificationType.Info,
+      });
+    } catch (e) {
+      handleError(e, 'Error restoring trash');
+    }
+  };
+</script>
+
+{#if $isMultiSelectState}
+  <AssetSelectControlBar assets={$selectedAssets} clearSelect={() => assetInteractionStore.clearMultiselect()}>
+    <SelectAllAssets {assetStore} {assetInteractionStore} />
+    <DeleteAssets force onAssetDelete={(assetId) => assetStore.removeAsset(assetId)} />
+    <RestoreAssets onRestore={(ids) => assetStore.removeAssets(ids)} />
+  </AssetSelectControlBar>
+{/if}
+
+{#if $featureFlags.loaded && $featureFlags.trash}
+  <UserPageLayout user={data.user} hideNavbar={$isMultiSelectState} title={data.meta.title}>
+    <div class="flex place-items-center gap-2" slot="buttons">
+      <LinkButton on:click={handleRestoreTrash}>
+        <div class="flex place-items-center gap-2 text-sm">
+          <HistoryOutline size="18" />
+          Restore All
+        </div>
+      </LinkButton>
+      <LinkButton on:click={() => (isShowEmptyConfirmation = true)}>
+        <div class="flex place-items-center gap-2 text-sm">
+          <DeleteOutline size="18" />
+          Empty Trash
+        </div>
+      </LinkButton>
+    </div>
+
+    <AssetGrid forceDelete {assetStore} {assetInteractionStore} removeAction={AssetAction.UNARCHIVE}>
```

Comment:
`removeAction` means "remove the asset from the store when this action happens". I think you don't need this. Or, if you do, it should be `AssetAction.RESTORE`. Basically, when you perma delete or restore an asset from multi-select it should get removed (no refresh required).